### PR TITLE
[Fix] Fix some warnings in unittest

### PR DESCRIPTION
**Pull request by teamwong111**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Fix some warnings in unittest.

## Modification

As above.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by zhouzaida** [comment]
Please fix the CI

**Comment by teamwong111** [comment]
> Please fix the CI

Done

File: tests/test_cnn/test_swish.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import torch
-from torch.nn.functional import sigmoid
+from torch import sigmoid
```
Comment (zhouzaida) [inline tests/test_cnn/test_swish.py:2]: import torch.nn.functional as F

F.sigmoid(...)

File: mmcv/ops/nms.py [hunk 2]
```diff
@@ -370,7 +370,7 @@ def nms_match(dets, iou_threshold):
     if isinstance(dets, torch.Tensor):
         return [dets.new_tensor(m, dtype=torch.long) for m in matched]
     else:
-        return [np.array(m, dtype=np.int) for m in matched]
+        return [np.array(m, dtype=int) for m in matched]
```
Comment (zhouzaida) [inline mmcv/ops/nms.py:373]: why we need to remove the `np.`

File: mmcv/ops/masked_conv.py [hunk 3]
```diff
@@ -62,8 +62,12 @@ def forward(ctx, features, mask, weight, bias, padding=0, stride=1):
...
                 pad_w=pad_w)
 
-            masked_output = torch.addmm(1, bias[:, None], 1,
-                                        weight.view(out_channel, -1), data_col)
+            masked_output = torch.addmm(
+                bias[:, None],
+                weight.view(out_channel, -1),
+                data_col,
+                beta=1,
+                alpha=1)
```
Comment (zhouzaida) [inline mmcv/ops/masked_conv.py:70]: We need to add a comment to explain why the order of parameters is changed.

File: tests/test_cnn/test_swish.py (hunk 1 above)
Comment (teamwong111) [reply tests/test_cnn/test_swish.py:2]: Is it any difference between torch.sigmoid and F.sigmoid?

File: mmcv/ops/nms.py (hunk 2 above)
Comment (teamwong111) [reply mmcv/ops/nms.py:373]: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

File: tests/test_cnn/test_swish.py (hunk 1 above)
Comment (zhouzaida) [reply tests/test_cnn/test_swish.py:2]: They are the same.

Comment (teamwong111) [reply tests/test_cnn/test_swish.py]: Done

File: mmcv/ops/masked_conv.py (hunk 3 above)
Comment (teamwong111) [reply mmcv/ops/masked_conv.py:76]: Done

File: mmcv/ops/masked_conv.py [hunk 4]
```diff
@@ -61,7 +61,13 @@ def forward(ctx, features, mask, weight, bias, padding=0, stride=1):
                 kernel_w=kernel_w,
                 pad_h=pad_h,
                 pad_w=pad_w)
-
+            # We change the order of parameters because:
+            # This overload of addmm is deprecated:
```
Comment (zhouzaida) [inline mmcv/ops/masked_conv.py:65]: Which PyTorch version deprecated the signature?

File: mmcv/cnn/utils/flops_counter.py [hunk 5]
```diff
@@ -503,8 +504,8 @@ def batch_counter_hook(module, input, output):
         batch_size = len(input)
     else:
         pass
-        print('Warning! No positional inputs found for a module, '
-              'assuming batch size is 1.')
+        warnings.warn('No positional inputs found for a module, '
+                      'assuming batch size is 1.')
```
Comment (zhouzaida) [inline mmcv/cnn/utils/flops_counter.py:508]: The above line is `pass` so will the warning be skipped?

Comment (teamwong111) [reply mmcv/cnn/utils/flops_counter.py:508]: No, `pass` is only a placeholder statement. Actually, I do not know why there is a `pass`.

File: mmcv/ops/masked_conv.py (hunk 4 above)
Comment (teamwong111) [reply mmcv/ops/masked_conv.py:65]: 1.5.0

File: mmcv/cnn/utils/flops_counter.py (hunk 5 above)
Comment (zhouzaida) [reply mmcv/cnn/utils/flops_counter.py:508]: OK, we can delete the pass

File: mmcv/ops/masked_conv.py (hunk 3 above)
Comment (zhouzaida) [reply mmcv/ops/masked_conv.py:76]: Although we do not provide the pre-built package for torch < 1.5.0 anymore, we also need to keep the compatible for them until next year.

Comment (teamwong111) [reply mmcv/ops/masked_conv.py:76]: Ok, I will give a WIP label to the pr and keep it open until we give up the compatibility for torch < 1.5.0.

File: mmcv/cnn/utils/flops_counter.py (hunk 5 above)
Comment (teamwong111) [reply mmcv/cnn/utils/flops_counter.py:507]: done

File: mmcv/ops/masked_conv.py (hunk 3 above)
Comment (teamwong111) [reply mmcv/ops/masked_conv.py]: done

**Review by zhouzaida — APPROVED** [review]

File: mmcv/cnn/bricks/generalized_attention.py [hunk 6]
```diff
@@ -131,7 +131,7 @@ def __init__(self,
 
             max_len_kv = int((max_len - 1.0) / self.kv_stride + 1)
             local_constraint_map = np.ones(
-                (max_len, max_len, max_len_kv, max_len_kv), dtype=np.int)
```
Comment (ZwwWayne) [inline mmcv/cnn/bricks/generalized_attention.py:134]: might need to ensure the lowest version requirement of numpy for this cleaning

Comment (teamwong111) [reply mmcv/cnn/bricks/generalized_attention.py:134]: I think it is safe because torch 1.3.0 corresponds to NumPy 1.21.2. However, `np.int` is a deprecated alias for the builtin `int` in NumPy 1.20.

**Review by ZwwWayne — APPROVED** [review]
